pyrealm/demography/canopy_functions.py

```diff
@@ -228,6 +139,38 @@ def calculate_relative_canopy_radius_at_z(
     return m * n * z_over_height ** (n - 1) * (1 - z_over_height**n) ** (m - 1)
 
 
+def calculate_crown_radius(
+    q_z: NDArray[np.float32],
+    r0: NDArray[np.float32],
+    validate: bool = True,
+) -> NDArray[np.float32]:
+    r"""Calculate crown radius from relative crown radius and crown r0.
+
+    The relative crown radius (:math:`q(z)`) at a given height :math:`z` describes the
+    vertical profile of the crown shape, but only varies with the ``m`` and ``n`` shape
+    parameters and the stem height. The actual crown radius at a given height
+    (:math:`r(z)`) needs to be scaled using :math:`r_0` such that the maximum crown area
+    equals the expected crown area given the crown area ratio traiit for the plant
+    functional type:
+
+    .. math::
+
+        r(z) = r_0 q(z)
+
+    This function calculates :math:`r(z)` given estimated ``r0`` and an array of
+    relative radius values.
+
+    Args:
+        q_z: TODO
+        r0: TODO
+        validate: Boolean flag to suppress argument validation.
+    """
+
+    # TODO - think about validation here. qz must be row array or 2D (N, n_pft)
```

Comment:
Could go into an issue, so that it's not forgotten

Reply:
I think I've one more chunk of stuff to pick out of #296  and then synchronizing the validation across these modules is the next thing to do. So I agree but it is coming imminently anyway 😄